test/unit/jwt-token-manager.test.js

```diff
@@ -175,19 +177,40 @@ describe('JWT Token Manager', () => {
     });
   });
 
+  describe('tokenNeedsRefresh', () => {
+    it('should return true if current time is past expire time', () => {
+      const instance = new JwtTokenManager();
+      instance.refreshTime = getCurrentTime() - 1000;
+
+      expect(instance.tokenNeedsRefresh()).toBe(true);
+    });
+
+    it('should return false if current time has not reached expire time', () => {
+      const instance = new JwtTokenManager();
+      instance.refreshTime = getCurrentTime() + 1000;
+
+      expect(instance.tokenNeedsRefresh()).toBe(false);
+    });
+
+    it('should return true if expire time has not been set', () => {
```

Comment:
I think this one is ok, because if the expire time hasn't been set the refresh time doesn't get set.